main.py

```
@@ -1,15 +1,73 @@
+                    await application.bot.set_webhook(url=WEBHOOK_URL, drop_pending_updates=True)
+                    logging.info("Webhook set successfully")
+                except Exception as e:
+                    logging.exception(f"Failed to set webhook: {e}")
+            else:
+                try:
+                    await application.bot.delete_webhook(drop_pending_updates=True)
+                    logging.info("Webhook deleted (no PUBLIC_BASE_URL configured)")
+                except Exception as e:
+                    logging.exception(f"Failed to delete webhook: {e}")
+        except Exception as e:
+            logging.error(f"Failed to initialize bot: {e}")
```

Ruff (0.13.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)